### PR TITLE
Addressing LFDA sign indeterminacy

### LINKED ISSUE #211: lfda not deterministic
See comment https://github.com/metric-learn/metric-learn/pull/193#pullrequestreview-245994079
It seems that `LFDA` is not deterministic because of an eigenvalue decomposition that can return one sign or another as output
More details to come

### PR DESCRIPTION
**Pull request by mvargas33**
Resolves #211 

I've looked for this issue and was able to replicate the random behavior when calculating the eigenvectors. Effectively it may happen that some eigenvectors might be signed-flipped, if you execute the code at different times.

None of the scipy functions we use to calculate the eigenvectors in LFDA supports something like a 'random_state': [scipy.linalg.eigh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eigh.html), [scipy.linalg.eig ](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eig.html)and [scipy.sparse.linalg.eigsh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigsh.html)

There are two alternatives:

1. We implement some kind of "convention" to force/determine the behavior. For instance, "Every first value of an eigenvector must be positive, otherwise flip the sign of the entire vector". (Flipping sign of an eigenvector, doesn't affect the eigenvector property)
2. Add a Note in documentation just like sklearn does for [SVD](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.TruncatedSVD.html)

Number 1 decreases performance a bit, and number 2 seems more reasonable. Consider the message I made a draft.

### CODE REVIEW

**Review by perimosocordiae — APPROVED** [review]
I agree that a documentation fix is the way to go here. Thanks!
